### ISSUE TITLE
non-selectable socket created

### ISSUE THREAD

**Issue by tiger5226** [lbryio/lbrycrd#367]
Still getting this inside containers for lbrycrd. I can't run lbrycrd in a container after initial sync. If the container goes down or if its stateless and is recreated, this error pops up and crashes lbrycrd. All versions 17.3.2.1 and up. 

```
lbrycrd_1  | 2020-02-20T04:17:31Z BerkeleyEnvironment::Open: LogDir=/data/.lbrycrd/database ErrorFile=/data/.lbrycrd/db.log
lbrycrd_1  | 2020-02-20T04:17:32Z [default wallet] nFileVersion = 170302
lbrycrd_1  | 2020-02-20T04:17:32Z [default wallet] Keys: 2002 plaintext, 0 encrypted, 2002 w/ metadata, 2002 total. Unknown wallet records: 1
lbrycrd_1  | 2020-02-20T04:17:32Z [default wallet] Wallet completed loading in              60ms
lbrycrd_1  | 2020-02-20T04:17:32Z [default wallet] setKeyPool.size() = 2000
lbrycrd_1  | 2020-02-20T04:17:32Z [default wallet] mapWallet.size() = 1
lbrycrd_1  | 2020-02-20T04:17:32Z [default wallet] mapAddressBook.size() = 1
lbrycrd_1  | 2020-02-20T04:17:32Z mapBlockIndex.size() = 741815
lbrycrd_1  | 2020-02-20T04:17:32Z nBestHeight = 719000
lbrycrd_1  | 2020-02-20T04:17:32Z Cannot create connection: non-selectable socket created (fd >= FD_SETSIZE ?)
lbrycrd_1  | 2020-02-20T04:17:32Z Error: Couldn't open socket for incoming connections (socket returned error No such file or directory (2))
lbrycrd_1  | 2020-02-20T04:17:32Z Cannot create connection: non-selectable socket created (fd >= FD_SETSIZE ?)
lbrycrd_1  | 2020-02-20T04:17:32Z Error: Couldn't open socket for incoming connections (socket returned error No such file or directory (2))
lbrycrd_1  | 2020-02-20T04:17:32Z Error: Error: Couldn't open socket for incoming connections (socket returned error No such file or directory (2))
lbrycrd_1  | Error: Error: Couldn't open socket for incoming connections (socket returned error No such file or directory (2))
lbrycrd_1  | Error: Failed to listen on any port. Use -listen=0 if you want this.
lbrycrd_1  | 2020-02-20T04:17:32Z Error: Failed to listen on any port. Use -listen=0 if you want this.
lbrycrd_1  | 2020-02-20T04:17:32Z tor: Thread interrupt
lbrycrd_1  | 2020-02-20T04:17:32Z Shutdown: In progress...
```

**Comment by nitronick600** [comment]
I'm getting this too.

**Comment by bvbfan** [comment]
How this can happen
https://github.com/lbryio/lbrycrd/blob/master/src/compat.h#L114
I suspect using an onld build or build that not include these changes

**Comment by nitronick600** [comment]
@bvbfan I'm just using the binary from the v0.17.3.2 release

**Comment by bvbfan** [comment]
Can you use v0.17.4.2 where the fix presents. It needs full reindex for SQLite backend.

**Comment by BrannonKing** [comment]
Don't use 17.4.2. It has too many bugs. I'll post a new build shortly.

**Comment by tiger5226** [comment]
Confirmed fix is good on 17.4.3 release
